### PR TITLE
fix: [GTM-638]: fix update LS default experience when switch default experience

### PR DESCRIPTION
**Pull request by nanaxuHarness**
Summary:
this is to fix when open a new tab after switch default experience, the new tab should go to new experience:

- module selection
- change default experience from account settings
- click Launch first generation

Jira Links:
https://harness.atlassian.net/browse/PLG-638

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/10-common/hooks/useUpdateLSDefaultExperience.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  const { accountId } = useParams<AccountPathProps>()
+  const defaultExperienceMap = currentUserInfo.accounts?.reduce((previousValue, account) => {
+    const newExperience = account.uuid === accountId ? experience : account.defaultExperience
+    return {
+      ...previousValue,
+      [account.uuid as string]: newExperience
+    }
+  }, {})
+
+  function updateLSDefaultExperience(): void {
```
[the remark this answers is not in the document]
Comment (nanaxuHarness) [reply src/modules/10-common/hooks/useUpdateLSDefaultExperience.ts]: passed the prop

File: src/modules/10-common/pages/welcome/ModuleInfo.tsx [hunk 2]
```diff
@@ -39,12 +40,16 @@ const ModuleInfo: React.FC<ModuleProps> = ({ module = 'cd' }) => {
...
+  const updatedDefaultExperience = !selectedInfoCard || selectedInfoCard?.isNgRoute ? Experiences.NG : Experiences.CG
+  const { updateLSDefaultExperience } = useUpdateLSDefaultExperience(updatedDefaultExperience)
 
   const getModuleLink = (moduleLinkArg: Module): React.ReactElement => {
     async function handleUpdateDefaultExperience(): Promise<void> {
       try {
         await updateDefaultExperience({
-          defaultExperience: !selectedInfoCard || selectedInfoCard?.isNgRoute ? Experiences.NG : Experiences.CG
+          defaultExperience: updatedDefaultExperience
+        }).then(() => {
```
[the remark this answers is not in the document]
Comment (nanaxuHarness) [reply src/modules/10-common/pages/welcome/ModuleInfo.tsx]: updated

File: src/modules/10-common/pages/welcome/ModuleInfo.tsx [hunk 3]
```diff
@@ -75,16 +80,12 @@ const ModuleInfo: React.FC<ModuleProps> = ({ module = 'cd' }) => {
         intent="primary"
         onClick={async () => {
           trackEvent(PurposeActions.CDCGModuleSelected, { category: Category.SIGNUP, module: ModuleName.CD })
-
-          await updateDefaultExperience({
-            defaultExperience: !selectedInfoCard || selectedInfoCard?.isNgRoute ? Experiences.NG : Experiences.CG
+          handleUpdateDefaultExperience().then(() => {
```
[the remark this answers is not in the document]
Comment (nanaxuHarness) [reply src/modules/10-common/pages/welcome/ModuleInfo.tsx]: updated

File: src/modules/10-common/hooks/useUpdateLSDefaultExperience.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    localStorage.setItem('defaultExperienceMap', JSON.stringify(defaultExperienceMap))
+  }
```
[the remark this answers is not in the document]
Comment (nanaxuHarness) [reply src/modules/10-common/hooks/useUpdateLSDefaultExperience.ts]: we don't necessarily update userInfo when switch experiences, for example, when we hit `launch 1st generation`, we just update the local cache default experience.

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by chrisblakely01 — APPROVED** [review]

**Pull request merged**